plugin.py

```diff
@@ -30,7 +30,11 @@ def plugin_info(self):
 
     def process(self, trace, data_context):
         with trace.open(data_type='text', mode='wb') as writer:
-            kaitai_utils.write_to_json(trace.open(), writer, kaitai_utils.get_kaitai_class())
+            kaitaiclass = kaitai_utils.get_kaitai_class()
+            with trace.open() as data:
+                kaitai_utils.write_to_json(data, writer, kaitaiclass)
+            with trace.open() as data:
+                kaitai_utils.create_bytearray_child_traces(data, kaitaiclass, trace, '$')
```

Comment:
Can you make an overload in the kaitai_utils that has no 'initial' string, this method can call the other method that is recursive. This $ looks very random. Maybe you can extract is as a constant in kaitai_utils. 'json_path_start_indicator' or something.

Reply:
For the sake of performance, maybe we can parse the struct in one pass (do all the magic in `write_to_json`)? It will make the code better maintainable as well (I hope), since there is just one object-traversal method instead of two).

(you can make the current json-path an argument with default value `$` so we don't have to pass it from here)

Reply:
In HANSKEN-21046, the magic happens in this function and the '$' is passed as default parameter.